src/purchase/utils.py

```diff
@@ -31,3 +39,59 @@ def store_leftover_paper_support(paper, purchase, leftover_amount):
         item=paper,
         hold_type=Escrow.AUTHOR_RSC,
     )
+
+
+def create_fundraise_with_escrow(
+    user, unified_document, goal_amount, goal_currency=USD, status=Fundraise.OPEN
+):
+    """
+    Helper function to create a fundraise with its associated escrow.
+    Returns (fundraise, error_response) tuple where error_response is None if successful
+
+    Note: This function should be called within a transaction.atomic() block
+    """
+    # Validate inputs
+    if unified_document.document_type != PREREGISTRATION:
+        return None, Response(
```

Comment:
This code mixes business logic with presentation logic. It would be cleaner to have the function return values or raise exceptions that are independent of the HTTP view layer. The fact that `rest_framework` is imported here is an indicator for inappropriately mixing these two aspects.